### PR TITLE
Added debugging support to services

### PR DESCRIPTION
**Pull request by mserranom**
Adds `--inspect` to node processes when `$DEBUG_NODE=="true"`
- Environment variable is passed via `docker-compose.debug.yml` (https://github.com/overleaf/overleaf/pull/634)
- Node<7 services (real time and filestore) don't support debugging since `--inspect` is not supported.
- Port binding is consistent with other debugging systems.

Please ignore changes in `settings.coffee`, it's all `.editorconfig` formatting.

### CODE REVIEW

File: runit/web-sharelatex/run [hunk 1]
```diff
@@ -1,3 +1,10 @@
 #!/bin/bash
 export SHARELATEX_CONFIG=/etc/sharelatex/settings.coffee
-exec /sbin/setuser www-data /usr/bin/node /var/www/sharelatex/web/app.js >> /var/log/sharelatex/web.log 2>&1
\ No newline at end of file
+
+NODE_PARAMS=""
+if [ "$DEBUG_NODE" == "true" ]; then
+    echo "running debug - web"
+    NODE_PARAMS="--inspect=0.0.0.0:40000"
```
Comment (das7pad) [inline runit/web-sharelatex/run:7]:
```suggestion
    NODE_PARAMS="--inspect=0.0.0.0:30000"
```

30000 would be consistent with the other ports

File: settings.coffee [hunk 2]
```diff
@@ -34,7 +34,7 @@ settings =
 	# Documentation about the URL connection string format can be found at:
 	#
 	#    http://docs.mongodb.org/manual/reference/connection-string/
-	#
```
Comment (das7pad) [inline settings.coffee:37]: unrelated whitespace changes

`$ git checkout upstream/master settings.coffee`

**Review by das7pad — CHANGES_REQUESTED** [review]

File: settings.coffee (hunk 2 above)
Comment (das7pad) [reply settings.coffee:37]: I just saw the update of the PR body.

File: runit/web-sharelatex/run (hunk 1 above)
Comment (mserranom) [reply runit/web-sharelatex/run:7]: 40000 for `web` is consistent with existing tooling already set in place.

Comment (das7pad) [reply runit/web-sharelatex/run:7]: That's internal I guess based on [1]?

---
[1] https://github.com/search?l=&q=4000+user%3Aoverleaf

**Review by gh2k — APPROVED** [review]
